sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateAsyncClientTest.java

```diff
@@ -630,24 +589,18 @@ public void listCertificates(HttpClient httpClient, CertificateServiceVersion se
         listCertificatesRunner((certificatesToList) -> {
             HashSet<String> certificates = new HashSet<>(certificatesToList);
 
-            for (String certName : certificates) {
-                PollerFlux<CertificateOperation, KeyVaultCertificateWithPolicy> certPoller =
-                    certificateAsyncClient.beginCreateCertificate(certName, CertificatePolicy.getDefault());
-
-                AsyncPollResponse<CertificateOperation, KeyVaultCertificateWithPolicy> lastResponse =
-                    certPoller.takeUntil(apr -> apr.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED)
-                        .blockLast();
-            }
-
-            sleepInRecordMode(90000);
-
-            List<CertificateProperties> output = new ArrayList<>();
-            StepVerifier.create(certificateAsyncClient.listPropertiesOfCertificates().map(certificate -> {
-                certificates.remove(certificate.getName());
-                return Mono.empty();
-            })).assertNext(ignore -> {
-                assertEquals(0, certificates.size());
-            });
+            StepVerifier.create(
+                    Flux.fromIterable(certificates)
+                        .map(certName -> certificateAsyncClient.beginCreateCertificate(certName, CertificatePolicy.getDefault())
+                            .takeUntil(apr -> apr.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED).last())
+                        .last().map(ignored -> certificateAsyncClient.listPropertiesOfCertificates()
+                            .map(certificate -> {
+                                certificates.remove(certificate.getName());
+                                return Mono.empty();
+                            })))
+                .assertNext(ignore -> {
+                    assertEquals(0, certificates.size());
+                });
```

Comment:
The recording for this test turned out empty because we are not waiting for the async operation to complete.
```suggestion
                }).verifyComplete();
```